cmd/buildah/build.go

```diff
@@ -88,342 +69,34 @@ func init() {
-	options := define.BuildOptions{
-		AddCapabilities:         iopts.CapAdd,
-		AdditionalTags:          tags,
-		AllPlatforms:            iopts.AllPlatforms,
-		Annotations:             iopts.Annotation,
-		Architecture:            systemContext.ArchitectureChoice,
-		Args:                    args,
-		AdditionalBuildContexts: additionalBuildContext,
-		BlobDirectory:           iopts.BlobCache,
-		CNIConfigDir:            iopts.CNIConfigDir,
-		CNIPluginPath:           iopts.CNIPlugInPath,
-		CommonBuildOpts:         commonOpts,
-		Compression:             compression,
-		ConfigureNetwork:        networkPolicy,
-		ContextDirectory:        contextDir,
-		CPPFlags:                iopts.CPPFlags,
-		DefaultMountsFilePath:   globalFlagResults.DefaultMountsFile,
-		Devices:                 iopts.Devices,
-		DropCapabilities:        iopts.CapDrop,
-		Err:                     stderr,
-		ForceRmIntermediateCtrs: iopts.ForceRm,
-		From:                    iopts.From,
-		IDMappingOptions:        idmappingOptions,
-		IIDFile:                 iopts.Iidfile,
-		In:                      stdin,
-		Isolation:               isolation,
-		IgnoreFile:              iopts.IgnoreFile,
-		Labels:                  iopts.Label,
-		Layers:                  layers,
-		LogRusage:               iopts.LogRusage,
-		Manifest:                iopts.Manifest,
-		MaxPullPushRetries:      maxPullPushRetries,
-		NamespaceOptions:        namespaceOptions,
-		NoCache:                 iopts.NoCache,
-		OS:                      systemContext.OSChoice,
-		Out:                     stdout,
-		Output:                  output,
-		BuildOutput:             iopts.BuildOutput,
-		OutputFormat:            format,
-		PullPolicy:              pullPolicy,
-		PullPushRetryDelay:      pullPushRetryDelay,
-		Quiet:                   iopts.Quiet,
-		RemoveIntermediateCtrs:  iopts.Rm,
-		ReportWriter:            reporter,
-		Runtime:                 iopts.Runtime,
-		RuntimeArgs:             runtimeFlags,
-		RusageLogFile:           iopts.RusageLogFile,
-		SignBy:                  iopts.SignBy,
-		SignaturePolicyPath:     iopts.SignaturePolicy,
-		Squash:                  iopts.Squash,
-		SystemContext:           systemContext,
-		Target:                  iopts.Target,
-		TransientMounts:         iopts.Volumes,
-		OciDecryptConfig:        decConfig,
-		Jobs:                    &iopts.Jobs,
-		Excludes:                excludes,
-		Timestamp:               timestamp,
-		Platforms:               platforms,
-		UnsetEnvs:               iopts.UnsetEnvs,
-		Envs:                    iopts.Envs,
-		OSFeatures:              iopts.OSFeatures,
-		OSVersion:               iopts.OSVersion,
-	}
-	if iopts.Quiet {
-		options.ReportWriter = ioutil.Discard
-	}
-
 	id, ref, err := imagebuildah.BuildDockerfiles(getContext(), store, options, containerfiles...)
 	if err == nil && options.Manifest != "" {
 		logrus.Debugf("manifest list id = %q, ref = %q", id, ref.String())
 	}
+	for _, f := range removeAll {
+		os.RemoveAll(f)
```

Comment:
Could use a `defer` here instead i think this entire loop should go just after we fetch `removeAll` in a `defer` call so we cleanup always otherwise if we return early because of `err` we will miss cleaning these files.